sympy/functions/elementary/trigonometric.py

```diff
@@ -117,10 +117,18 @@ def _peeloff_pi(arg):
             K, p = a.as_two_terms()
             if p is S.Pi and K.is_Rational:
                 break
+            K = a.coeff(S.Pi)
+            if K and K.is_rational:
+                break
     else:
         return arg, S.Zero
 
-    m1 = (K % S.Half)*S.Pi
+
+    z = S.One
+    for a in Mul.make_args(K):
+        z = z*(a % S.Half)
```

Comment:
This might do something unexpected if `a` is complex. See https://github.com/sympy/sympy/issues/11391.

Reply:
Anyway I don't see how this is different from what was already there. If Mod doesn't properly distribute over multiplication it should be fixed there.

Reply:
Yes it does not in my opinion.
Because (2*n mod S. Half) returns 2nmod(1/2) instead of returning 0 and due to this the original issue was occurring.

Reply:
Where n is an integer

Reply:
> This might do something unexpected if `a` is complex. See #11391.

Will enter an "if" condition so that this loop runs only when K is rational.